test/trigger/test_trigger.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+// This test loads an individual trigger file and makes validates
+// the format and regex calls made.
+
+// TODO: Remove ` ?` before each hex value once global prefix `^.{14} ` is added.
+// JavaScript doesn't allow for possessive operators in regular expressions.
+
+let fs = require('fs');
+
+let exitCode = 0;
+
+let inputFilename = String(process.argv.slice(2));
+
+let testValidTriggerRegexLanguage = function(file, contents) {
+  let unsupportedRegexLanguage = /(?:regex|triggerRegex)(?!:|Cn|De|Fr|Ko|Ja).*?:/g;
```

Comment:
This is neither here nor there, but I wonder a little bit about the inconsistency between infoText (and friends) returning a dict indexed by lang, and regex having different keys entirely.  <_<

Reply:
I think the infoText approach is possibly better? If nothing else, it might clean up the JavaScript logic in `popup-text.js`; I tried adding a mock version of `triggerRegex` to the logic [found here](https://github.com/quisquous/cactbot/blob/master/ui/raidboss/popup-text.js#L119), and found that it got a little obnoxious looking for 4 potential variables (triggerRegexFr, triggerRegex, regexFr, regex in one example), with an inherent pecking order. Although, I can't say I've tried the opposite; and, I don't know how much better it would be dealing with 3 variables (triggerRegex[fr], regexFr, regex) over 4. But, I think if you were talking end-state, it might be better to go with the dict, how do you feel about it?

Reply:
Yeah, I think it would.  I think consistency would be nice here too to not have two ways to specify translations.  There is a pecking order inherent in the keys as well, so I don't think that would improve.

(I was commenting about consistency in oopsy vs raidboss translations in #619, which is another point of inconsistency.  I think I was trying something new there and I'm not sure that it works out in the end.  Translated effect names make a lot more sense in ui/jobs but maybe less sense in a triggers-esque file.)

I haven't really deprecated much in cactbot, but at some point I think it would make sense to switch from regexFr to regex['fr'].  Probably very few people have user triggers that are translated? (And if we're deprecating things maybe somebody we can change everything on `data` to have proper javascript camelCasing <_<)

This sort of thing doesn't really rise to the level of putting in a github issue (or maybe it does?), but I'm not quite sure where to put these sorts of thoughts for the long term, so it's clear like "I'm not happy about this, but it'd be a bunch of work to change".